### ISSUE TITLE
预训练

### ISSUE THREAD

**Issue by imfinethanks** [ZiweiWangTHU/BiDet#59]
作者你好，这个代码库是我们能找到的唯一一个开源二值化目标检测代码库，为我们的工作提供了巨大的帮助，十分感谢。
我们想在这个代码库的基础上改进模型，但这就需要对模型进行重新预训练，而这个代码库直接提供了预训练完的权重文件而不是预训练的代码。为了公平比较，我们希望在改进模型后能采用和你们一致的预训练方法，不知道是否方便提供模型的预训练代码？

**Comment by Wuziyi616** [comment]
这里的预训练是指 VGG/ResNet 吗？我现在已经不做二值网络了，但我记得 binary classification network 应该是有不少开源项目的，或许你可以直接使用它们的 ImageNet 训练代码，我这里用的也就是最简单的 cross-entropy loss 训练

**Comment by imfinethanks** [comment]
好的 谢谢

**Comment by imfinethanks** [comment]
作者你好，我使用recu的代码重新预训练了bidet的backbone，使用的也是最简单的cross-entropy loss，并在imagenet上达到了56的top-1精度，高于你提供的预训练权重的top-1精度54。但是使用了预训练的权重后，bidet的map大幅度下降从59.3降到了40左右。
请问重新预训练backbone和使用重新预训练的backbone需要注意什么？可能是什么原因导致了同样的模型预训练时的精度更高，但权重给到bidet后map却大幅下降呢？

**Comment by Wuziyi616** [comment]
hummm 我不大确定，我当时应该只做过两个 VGG 结构的调整：就是增加了一些 residual connection + 减少了一些 downsampling 操作 (比如 pooling 或者 Conv stride). 你可以确认一下你的 VGG 结构是不是和我的一样？

**Comment by imfinethanks** [comment]
我训的是faster_rcnn，voc数据集，backbone是resnet18。

**Comment by Wuziyi616** [comment]
在我这没出现过这种情况, 我记得 Faster R-CNN 一般挺稳定的; 用原来我的 pre-trained weight 你能复现结果吗？这可以确认你的环境/data 没有问题；如果可以的话 那我确实想不出别的原因了

**Comment by Wuziyi616** [comment]
哦还有一种可能, 你这里提到的 recu 是另一个 binary network 吗？有没有可能他的 weight 直接 load 进我的 [Bi-Real ResNet](https://github.com/ZiweiWangTHU/BiDet/blob/master/faster_rcnn/lib/model/faster_rcnn/bidet_resnet.py) 不兼容导致性能下降？

**Comment by imfinethanks** [comment]
用你的权重可以复现，recu我只用了它的训练代码，网络结构还是从bidet直接复制过去的，很奇怪
